### ISSUE TITLE
Resolve issue with Android code provided on 'Writing custom platform-specific code' page

### ISSUE THREAD

**Issue by 0xcodezero** [flutter/flutter#148904]
Page URL: https://docs.flutter.dev/platform-integration/platform-channels
Page source: https://github.com/flutter/website/tree/master/src/docs/development/platform-integration/platform-channels.md

Description of issue:

Currently, this code is not working anymore for Android (Java or Kotlin) as mostly there is no more method `onCreate` is there, and the `MainActivity` type has been changed to use 

```
import io.flutter.embedding.android.FlutterActivity;
import io.flutter.embedding.engine.FlutterEngine;
```

instead of 
`import io.flutter.app.FlutterActivity;`

And I got an error for `getFlutterView() is not defined`
Any Suggestion?

**Comment by AbdeMohlbi** [comment]
AFAIK the examples are correct now, i think this was fixed in https://github.com/flutter/website/pull/3438 
@reidbaker can you confirm ?

**Comment by reidbaker** [comment]
I dont think so the back and forth with @mossmana happened well after that pr you linked landed. Transfering to @jwill who has some related work for platform views.
